ReadMe.rst

```diff
@@ -0,0 +1,91 @@
+platform firmware must implement to satisfy the relevant API requirements defined in the UEFI Specification. While the
+underlying implementation is platform-specific, the callers will include both the operating system and firmware
+components.
+
+Motivation
+==========
+
+The de facto open-source implementation of UEFI, `TianoCore`_, provides a commonly used set of `UEFI variable drivers`_
+in the `edk2`_ project that has served as the industry standard implementation for UEFI variable services for over a
+decade. Over time, the UEFI variable driver has substantially grown in complexity to support an increasing number of
+features.
+
+The TianoCore driver is now over 15 years old. It's design is rigid and not accommodating to change. Over the span of
+its lifetime, many advancements have occurred in the PC industry that require better scale to support:
+
+1. New storage technologies have come to market
+2. Device trends have shifted to low-power ultra mobile devices and cloud server systems
+3. New offload engines like BMC and special security processors have become more common to process non-volatile data
+4. New expectations around device security have come into focus
+
+   - For example, resistance against physical attack has led to variable data confidentiality via encryption, data
+     integrity checks for tamper-proof storage guarantees, data replay protection, etc.
+5. Additional computer architectures have gained popularity such as AArch64 and RISC-V
+6. Operating systems have evolved and so have their security expectations
+
+The TianoCore driver was written for a PI-centric boot flow assuming it was writing to SPI flash with no structured
+design to support extending the driver to support these advancements. In addition, while some industry standard tests
+such as the UEFI Self-Certification Tests exist, much of the stack is error prone to modify and difficult to assess
+because of its accumulated technical debt.
+
+Due to the importance of the driver, we concluded that a new design that takes into account these requirements with
+testing built in could better support today's needs.
+
+Code of Conduct
+===============
+
+This project has adopted the Microsoft Open Source Code of Conduct https://opensource.microsoft.com/codeofconduct/
+
+For more information see the Code of Conduct FAQ https://opensource.microsoft.com/codeofconduct/faq/
+or contact `opencode@microsoft.com <mailto:opencode@microsoft.com>`_. with any additional questions or comments.
+
+Builds
+======
+
+Please follow the steps in the Project Mu docs to build for CI and local testing.
+`More Details <https://microsoft.github.io/mu/CodeDevelopment/compile/>`_
+
+Copyright & License
+===================
+
+Some files in this repository have their own copyright. Otherwise, the following copyright applies.
+
+| Copyright (C) Microsoft Corporation
+| SPDX-License-Identifier: BSD-2-Clause-Patent

```

Comment:
in my view using github you didn't get a line break between these two lines and it doesn't look good.

Reply:
I agree the "rich diff" doesn't look great. Does it look better rendered here? https://github.com/makubacki/mu_feature_uefi_variable/tree/add_readme_rst

It looks fine to me outside the diff view.

Reply:
I'm going to go ahead and merge so we have a readme file. I'm happy to make any adjustments, just let me know.